app/components/phone_input_component.rb

```diff
@@ -21,15 +21,26 @@ def initialize(
   end
 
   def supported_country_codes
-    codes = PhoneNumberCapabilities::INTERNATIONAL_CODES.keys
-    codes &= allowed_countries if allowed_countries
-    codes
+    @supported_country_codes ||= begin
+      codes = PhoneNumberCapabilities::INTERNATIONAL_CODES.keys
+      codes &= allowed_countries if allowed_countries
+      codes
+    end
+  end
+
+  def translated_country_code_names
+    return {} if I18n.locale == :en
+
```

Comment:
Is there a strong reason _not_ to translate the country names for English? While probably unnecessary, it would be good to consistently use our locale data as the source of truth for the option texts.

Reply:
If we provide localized translations for English, the above feature will be overridden.